### PR TITLE
Allow product tab navigation without prompting for unsaved changes

### LINKED ISSUE #36205: Product tabs trigger unsaved changes warning on navigation
When navigating between product tabs, the unsaved work warning is triggered.  Since the tabs are not a true navigation away from the page, they should be navigable without this prompt.

### Screenshot

Image

### Acceptance criteria

* [ ] Navigating between tabs should not trigger the warning
* [ ] Navigating to other pages away from the form should still trigger the warning

### PR DESCRIPTION
**Pull request by joshuatf**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Previously, attempting to navigate between tabs on the same product form would trigger the unsaved changes warning if any form changes existed.  This PR allows navigation between the tabs without that warning.

Closes #36205  .

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Navigate to the new product experience (Products -> Add new (MVP))
2. Add some changes to the form, but don't save
3. Click on the tabs (General/Inventory/Shipping/etc)
4. Make sure you can navigate without being prompted to save changes
5. Click to any other page (try non-React pages such as legacy pages and React pages such as Analytics)
6. Make sure you are still prompted to save changes before leaving

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [x] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by AnnaMag — DISMISSED** [review]
LGTM and test well.

File: plugins/woocommerce-admin/client/hooks/usePreventLeavingPage.ts [hunk 1]
```diff
@@ -20,7 +26,9 @@ export default function usePreventLeavingPage(
 			__( 'Changes you made may not be saved.', 'woocommerce' ),
 		[ message ]
 	);
-	const { navigator } = useContext( NavigationContext );
+	const nav = useContext( NavigationContext );
+	const { navigator } = nav;
```
Comment (octaedro) [inline plugins/woocommerce-admin/client/hooks/usePreventLeavingPage.ts]: Nitpick: Is there a reason to change these lines? I'm not sure if it's worth adding the extra line.

**Review by octaedro — DISMISSED** [review]
Great job @joshuatf! This is testing well, and the code looks good to me. I only left a small suggestion, but I'll pre-approve this PR.

**Comment by joshuatf** [comment]
Hey @octaedro, just addressed the nitpick you mentioned in https://github.com/woocommerce/woocommerce/pull/36235/commits/47d70a73ffa732b4bb57edd11dcf2ede008112dc.  Could you give this another ✅ ?

**Review by octaedro — APPROVED** [review]
LGTM 🚀

**Pull request merged**